### DISCUSSION TITLE
Support for Direct Boot / alarm over the first unlock screen

### DISCUSSION THREAD

**Discussion by secretmango** [WrichikBasu/ShakeAlarmClock#67] [Feature requests]
The stock AOSP camera supports direct boot. This means that if a system update for example rebooted your device, the alarm will still show even if the device is still completely locked.

On GrapheneOS there is auto-reboot, a security measurement to clear the encryption keys from RAM.

The AOSP clock and [new-clock](https://github.com/qw123wh/new-clock) both support direct boot.

It could be nice having this functionality, alternatively GrapheneOS users have their preinstalled alarm without shake function, new-clock has the shake function too though.

[More info here](https://github.com/yuriykulikov/AlarmClock/issues/398)

**Comment by WrichikBasu** [comment]
This has been implemented in [v4.1.0-beta](https://github.com/WrichikBasu/ShakeAlarmClock/releases/tag/v4.1.0-beta).
